Source/core/DataElementFile.cpp

```diff
@@ -46,6 +46,8 @@ namespace Core {
         // The file needs to be prepared in the same way as we request the Memorymapped file...
         ASSERT(m_File.IsOpen() == false);
 
+        m_File.Open((type & File::USER_WRITE) == 0);
```

Comment:
This is changing the interface behaviour. It now opens the file, if it was not yet open. We need to eleborate a bit on this (probably open in the MessageDispatchr) to keep the default bahaviour like it was.

Reply:
In this line ``` ASSERT(m_File.IsOpen() == false);```` an assertion is performed to check if file is not open. Then, the function in the next line ```if (IsValid()) ``` internally checks if the file is open. Shouldn't the file be opened between those checks?